File: Launch/Mapping.py
import rclpy
from rclpy.node import Node
from sensor_msgs.msg import LaserScan
from nav_msgs.msg import OccupancyGrid

class MappingNode(Node):
    def __init__(self):
        super().__init__('mapping_node')
        self.subscription = self.create_subscription(
            LaserScan,
            'laser_scan_topic',  # Replace with your laser scan topic
            self.scan_callback,
            10)
        self.publisher = self.create_publisher(
            OccupancyGrid,
            'map_topic',  # Replace with your map topic
            10)

    def scan_callback(self, msg):
        # Process the laser scan data and update the map
        # Implement SLAM algorithms or use existing packages like gmapping

        # Publish the updated map
        map_msg = OccupancyGrid()
        # Populate the map_msg with your map data
        # Set map_msg.header, map_msg.info, and map_msg.data fields accordingly
        self.publisher.publish(map_msg)

def main(args=None):
    rclpy.init(args=args)
    mapping_node = MappingNode()
    rclpy.spin(mapping_node)
    mapping_node.destroy_node()
    rclpy.shutdown()

if __name__ == '__main__':
    main()
  

File: Launch/Mapping.cpp
#include "rclcpp/rclcpp.hpp"
#include "sensor_msgs/msg/laser_scan.hpp"
#include "nav_msgs/msg/occupancy_grid.hpp"

class MappingNode : public rclcpp::Node {
public:
  MappingNode() : Node("mapping_node") {
    subscription_ = create_subscription<sensor_msgs::msg::LaserScan>(
        "laser_scan_topic", 10, std::bind(&MappingNode::scan_callback, this, std::placeholders::_1));
    publisher_ = create_publisher<nav_msgs::msg::OccupancyGrid>("map_topic", 10);
  }

private:
  void scan_callback(const sensor_msgs::msg::LaserScan::SharedPtr msg) {
    // Process the laser scan data and update the map
    // Implement SLAM algorithms or use existing packages like gmapping

    // Publish the updated map
    auto map_msg = std::make_unique<nav_msgs::msg::OccupancyGrid>();
    // Popular
for slam mapping ;
